src/ui/api/admin_job.go

```diff
@@ -0,0 +1,178 @@
+
+	ajr := models.AdminJobReq{}
+	aj.DecodeJSONReq(&ajr)
+
+	aj.submitJob(ajr)
+}
+
+//Put ...
+func (aj *AdminJobAPI) Put() {
+	ajr := models.AdminJobReq{}
+	aj.DecodeJSONReq(&ajr)
+
+	query := &common_models.AdminJobQuery{
+		Name: ajr.Name,
+	}
+	jobs, err := dao.GetAdminJobs(query)
+	if err != nil {
+		aj.HandleInternalServerError(fmt.Sprintf("%v", err))
+		return
+	}
+	if len(jobs) != 1 {
+		log.Warningf("find more than one admin jobs.")
+	}
+
+	// stop the scheduled job and remove it.
+	for _, j := range jobs {
+		if err = utils_ui.GetJobServiceClient().PostAction(j.UUID, job.JobActionStop); err != nil {
+			if e, ok := err.(*common_http.Error); !ok || e.Code != http.StatusNotFound {
+				aj.HandleInternalServerError(fmt.Sprintf("%v", err))
+				return
+			}
+		}
+		if err = dao.DeleteAdminJob(j.ID); err != nil {
+			aj.HandleInternalServerError(fmt.Sprintf("%v", err))
+			return
+		}
+	}
+
+	aj.submitJob(ajr)
+}
+
+// Get ...
+func (aj *AdminJobAPI) Get() {
+	jobName := aj.GetStringFromPath(":name")
+	if len(jobName) <= 0 {
+		aj.HandleInternalServerError(fmt.Sprintf("need to specify job name"))
+		return
+	}
+
+	jobs, err := dao.GetAdminJobs(&common_models.AdminJobQuery{
+		Name: jobName,
+	})
+	if err != nil {
+		aj.HandleInternalServerError(fmt.Sprintf("failed to get admin jobs: %v", err))
+		return
+	}
+	aj.Data["json"] = jobs
+	aj.ServeJSON()
+}
+
+//GetLog ...
+func (aj *AdminJobAPI) GetLog() {
+	id, err := aj.GetInt64FromPath(":id")
+	if err != nil {
+		aj.HandleBadRequest("invalid ID")
+		return
+	}
+	job, err := dao.GetAdminJob(id)
+	if err != nil {
+		log.Errorf("Failed to load job data for job: %d, error: %v", id, err)
+		aj.CustomAbort(http.StatusInternalServerError, "Failed to get Job data")
+	}
+	if job == nil {
+		log.Errorf("Failed to get admin job: %d", id)
+		aj.CustomAbort(http.StatusInternalServerError, "Failed to get Job")
+	}
+
+	logBytes, err := utils_ui.GetJobServiceClient().GetJobLog(job.UUID)
+	if err != nil {
+		if httpErr, ok := err.(*common_http.Error); ok {
+			aj.RenderError(httpErr.Code, "")
+			log.Errorf(fmt.Sprintf("failed to get log of job %d: %d %s",
+				id, httpErr.Code, httpErr.Message))
+			return
+		}
+		aj.HandleInternalServerError(fmt.Sprintf("Failed to get job logs, uuid: %s, error: %v", job.UUID, err))
+		return
+	}
+	aj.Ctx.ResponseWriter.Header().Set(http.CanonicalHeaderKey("Content-Length"), strconv.Itoa(len(logBytes)))
+	aj.Ctx.ResponseWriter.Header().Set(http.CanonicalHeaderKey("Content-Type"), "text/plain")
+	_, err = aj.Ctx.ResponseWriter.Write(logBytes)
+	if err != nil {
+		aj.HandleInternalServerError(fmt.Sprintf("Failed to write job logs, uuid: %s, error: %v", job.UUID, err))
+	}
+}
+
+// submitJob submits a job to job service per request
+func (aj *AdminJobAPI) submitJob(ajr models.AdminJobReq) {
+	id, err := dao.AddAdminJob(&common_models.AdminJob{
+		Name: ajr.Name,
+		Kind: ajr.Kind,
+	})
+	if err != nil {
+		aj.HandleInternalServerError(fmt.Sprintf("%v", err))
```

Comment:
Should we update the job status in the database if got error?

Reply:
Should consider the scenario that users submit the GC job more than once, and how can we determine the status of the GC process?